sql/sql_update.cc

```diff
@@ -49,6 +49,335 @@
+    start_field->set_has_explicit_value();
+  }
+
+  if (likely(!res) && rcond)
+  {
+    res= period_conds.end.item->save_in_field(end_field, true);
+    end_field->set_has_explicit_value();
+  }
+
+  return res;
+}
+
+static
+int process_cut_fields_for_portion_of_time(TABLE* table)
+{
+  return cut_fields_for_portion_of_time(
+    table->in_use, 
+    table,
+    table->pos_in_table_list->period_conditions
+  );
+}
+
+static
+int process_update_row(TABLE* table)
+{
+  return table->file->ha_update_row(table->record[1],
+                                    table->record[0]);
+}
+
+static
+int process_triggers_before(THD *thd, TABLE *table,
+                           List<Item> *fields,
+                           List<Item> *values
+                          )
+{
+  bool trg_skip_row = false;
+  if (table->triggers->process_triggers(thd, TRG_EVENT_UPDATE, TRG_ACTION_BEFORE,
+                                  true, &trg_skip_row, fields) ||
+      not_null_fields_have_null_values(table))
+  {
+    return 1;
+  }
+
+  /*
+    Re-calculate virtual fields to cater for cases when base columns are
+    updated by the triggers.
+  */
+
+  if (table->vfield && fields->elements &&
+    no_need_to_skip_a_row(&trg_skip_row))
+  {
+    Item *fld= (Item_field*) fields->head();
+    Item_field *item_field= fld->field_for_view_update();
+    if (item_field)
+    {
+      DBUG_ASSERT(table == item_field->field->table);
+      return table->update_virtual_fields(table->file,
+                                            VCOL_UPDATE_FOR_WRITE);
+    }
+  }
+
+  if (trg_skip_row) 
+    return HA_ERR_SKIP_ROW;
+
+  return 0;
+}
+
+static
+void process_vers_update_end(TABLE* table) 
+{
+  table->vers_update_end();
+}
+
+static
+int process_vers_insert_history(TABLE* table, ha_rows* rows_inserted)
+{
+  store_record(table, record[2]);
+  table->mark_columns_per_binlog_row_image();
+  int error= vers_insert_history_row(table);
+  restore_record(table, record[2]);
+  if (likely(!error))
+    (*rows_inserted)++;
+  
+  return error;
+}
+
+static
+int process_check_view_conds(TABLE* table, ha_rows* found, bool* ignore)
+{
+  int res= table->pos_in_table_list->view_check_option(table->in_use, *ignore);
+
+  if (res != VIEW_CHECK_OK)
+    (*found)--;
+
+  return res;
+}
+
+static
+int period_make_inserts(TABLE* table, THD* thd, ha_rows* rows_inserted)
+{
+  store_record(table, record[2]);
+  restore_record(table, record[1]);
+  int error= table->insert_portion_of_time(
+    thd,
+    table->pos_in_table_list->period_conditions,
+    rows_inserted
+  );
+  restore_record(table, record[2]);
+
+  return error;
+}
+
+static
+int process_fill_record(THD* thd, TABLE* table,
+                        List<Item> *fields,
+                        List<Item> *values)
+{
+  return fill_record(thd, table, *fields, *values, false, true);
+}
+
+static
+int process_dec_limit_update(TABLE* table, ha_rows* limit, ha_rows* dup_key_found, 
```

Comment:
We'd beeter not have this much changes to the update function. Better to extract the function in such a way, that we'll see all changes are just mechanical. `process_dec_limit_update` and others are something new in scope of this patchset, making the changes more prono to mistakes